contracts/utils/SafeAccessControlEnumerable.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+import "./PendingOwnable.sol";
+
+error SafeAccessControlEnumerable__SenderMissingRoleAndIsNotOwner(
+    bytes32 role,
+    address sender
+);
+
+error SafeAccessControlEnumerable__RoleIsDefaultAdmin();
+
+abstract contract SafeAccessControlEnumerable is
+    PendingOwnable,
+    AccessControlEnumerable
+{
+    /**
+     * @dev Modifier that checks that the role is not the `DEFAULT_ADMIN_ROLE`
+     */
+    modifier roleIsNotDefaultAdmin(bytes32 role) {
+        if (role == DEFAULT_ADMIN_ROLE)
+            revert SafeAccessControlEnumerable__RoleIsDefaultAdmin();
+        _;
+    }
+
+    /**
+     * @dev Modifier that checks that an account is the `owner` or has a specific role
+     */
+    modifier onlyOwnerOrRole(bytes32 role) {
+        if (msg.sender != owner() || !hasRole(role, msg.sender))
```

Comment:
did you mean to do `&&` rather than `||`?

Reply:
lol, indeed